Makefile

```diff
@@ -0,0 +1,23 @@
+CC := gcc
+CFLAGS_BENCH := -std=gnu99 -O3
+LFLAGS_BENCH := -lpng
+CFLAGS_CONV := -std=c99 -O3
+
+TARGET_BENCH := qoibench
+TARGET_CONV := qoiconv
+
+all: $(TARGET_BENCH) $(TARGET_CONV)
+
+bench: $(TARGET_BENCH)
+
+$(TARGET_BENCH):$(TARGET_BENCH).c $(LFLAGS_BENCH)
+	$(CC) $(CFLAGS_BENCH) $(TARGET_BENCH).c -o $(TARGET_BENCH) $(LFLAGS_BENCH)
+
+conv: $(TARGET_CONV)
+$(TARGET_CONV):$(TARGET_CONV).c
+	$(CC) $(CFLAGS_CONV) $(TARGET_CONV).c -o $(TARGET_CONV)
+
+.PHONY: clean
+clean:
+	$(RM) $(TARGET_BENCH) $(TARGET_CONV) $(OBJS) $(DEPS)
+-include $(DEPS)
```

Comment:
`$(DEPS)` is undefined and none of the commands above even produce dependency files (with e.g. `-MD` or `-MMD`).

Reply:
Thanks, nice to know.
The "solution" would be to remove ``` $(OBJS) ``` and  ``` $(DEPS) ``` tags from the clean rule !?

Reply:
If you have a list of intermediate objects to create, you can usually do `$(patsubst …)` to transform the `.c` filenames into their `.o` object file equivalents. That way you still could clear the object files if you wanted to name them individually.

Alternatively many Makefiles also shortcut on using `${RM} -f *.o` and `${RM} -f *.d` (for the dependency files).

For the meantime dropping these two variables seems fine to me too.